imports/plugins/included/shipping-rates/server/hooks/hooks.js

```diff
@@ -58,21 +78,21 @@ function getShippingRates(previousQueryResults, cart) {
     merchantShippingRates = marketplaceSettings.public.merchantShippingRates;
   }
 
-  // TODO: Check to see if the merchantShippingRates flag is set in
-  // marketplace and get rates from the correct shop.
-  const pkgData = Packages.findOne({
-    name: "reaction-shipping-rates",
-    shopId: Reaction.getPrimaryShopId()
-  });
+  let pkgData;
+  if (merchantShippingRates) {
+    pkgData = Packages.findOne({
```

Comment:
Actually that's a good point since we need to not use the "current shop" we need to use the shops supplied in `cart.items`. I will add a TODO as there is a lot more to do here.

Reply:
I wonder if it wouldn't be better to throw some kinds of "not implemented" error here rather than an implementation that doesn't work

Reply:
I do think not implemented might be a good option. This gets into the need to have a workflow for making sure that shops only have their items published in a marketplace if they are all set up. Discussed in a little more detail here https://github.com/reactioncommerce/reaction/issues/2864#issuecomment-333272623